src/main/java/com/example/algorithm/QVoterModel.java

```diff
@@ -39,15 +44,12 @@ public void loadEnvironment(MyGraph<Integer, Integer> graph, AlgorithmSettings s
     public StepReport step() {
         switch (algorithmPhase){
             case SEND -> {
-                time ++;
                 algorithmPhase = QVoterModel.AlgorithmPhase.CHOOSE;
                 return sendOpinions();
             }
             case CHOOSE -> {
                 algorithmPhase = QVoterModel.AlgorithmPhase.SEND;
-                if(time == maxTime){
```

Comment:
I would not delete this condition, because now i we don't call explicitly checkIsFinished(), algorithm won't change it's finished state.

Reply:
I explained below why I decided to delete it.